File: BotsSDK/korebot/src/main/java/kore/botssdk/models/BotRequest.java
package kore.botssdk.models;

import kore.botssdk.net.RestResponse;

/**
 * Created by Pradeep Mahato on 03-Jun-16.
 * Copyright (c) 2014 Kore Inc. All rights reserved.
 */

public class BotRequest extends BaseBotMessage {

    private RestResponse.BotMessage message;
    private final String resourceid = "/bot.message";
    private BotInfoModel botInfo;
    private final long id = 1;

    public void setMessage(RestResponse.BotMessage message) {
        this.message = message;
    }

    public RestResponse.BotMessage getMessage() {
        return message;
    }

    public BotInfoModel getBotInfo() {
        return botInfo;
    }

    @Override
    public boolean isSend() {
        return true;
    }

    public void setCreatedOn(String createdOn) {
        this.createdOn = createdOn;
    }

}

File: BotSdk/botclient/src/main/java/com/kore/model/BotRequest.kt
package com.kore.model

data class BotRequest(
    val message: BotMessage?,
    val botInfo: BotInfoModel?,
    val id: Long,
    override val messageId: String = "",
    override val timeMillis: Long = 0,
    override val isSend: Boolean = true,
    override val from: String? = null,
    override val createdOn: String = "",
    override val formattedTime: String = "",
    override val messageDate: String = "",
    private val resourceId: String? = "/bot.message"
) : BaseBotMessage {
    fun updateRenderMessage() {
        message?.checkAndUpdateRenderMessage()
    }
}